README.md

```diff
@@ -35,8 +35,180 @@ import module from 'my_private_module'
+
+The configuration will be the same as the `module-alias.config.js` but a different file name.
+
+```js
+require('module-alias')({
+  base: 'path/to/the/configuration/file.js'
+})
+```
+
+If the `module-alias.config.js` file or your `package.json` file is in a sub-directory, simply pass the path to the directory in options.
+
+```js
+require('module-alias')({
+  base: 'path/to/the/sub-directory'
+})
+```
+
+If you prefer to set it all up programmatically, then the following methods are available for you:
+
+* `addAlias('alias', 'target_path')` - register a single alias
+* `addAliases({ 'alias': 'target_path', ... }) ` - register multiple aliases
+* `addPath(path)` - Register custom modules directory (like node_modules, but with your own modules)
+
+_Examples:_
+```js
+const moduleAlias = require('module-alias')
+
+//
+// Register alias
+//
+moduleAlias.addAlias('@client', __dirname + '/src/client')
+
+// Or multiple aliases
+moduleAlias.addAliases({
+  '@root'  : __dirname,
+  '@client': __dirname + '/src/client',
+  ...
+})
+
+// Custom handler function (starting from v2.1)
+moduleAlias.addAlias('@src', (fromPath, request, alias) => {
+  // fromPath - Full path of the file from which `require` was called
+  // request - The path (first argument) that was passed into `require`
+  // alias - The same alias that was passed as first argument to `addAlias` (`@src` in this case)
+
+  // Return any custom target path for the `@src` alias depending on arguments
+  if (fromPath.startsWith(__dirname + '/others')) return __dirname + '/others'
+  return __dirname + '/src'
+})
+
+//
+// Register custom modules directory
+//
+moduleAlias.addPath(__dirname + '/node_modules_custom')
+moduleAlias.addPath(__dirname + '/src')
+
+//
+// Import settings from a specific package.json
+//
+moduleAlias(__dirname + '/package.json')
+
+// Or let module-alias to figure where your package.json is
+// located. By default it will look in the same directory
+// where you have your node_modules (application's root)
+moduleAlias()
+```
+
+## Usage with WebPack
+
+Luckily, WebPack has a built in support for aliases and custom modules directories so it's easy to make it work on the client side as well!
+
+### With module-alias.config.js
+
+```js
+// webpack.config.js
+const config = require('./module-alias.config.js')
+
+module.exports = {
+  entry: { ... },
+  resolve: {
+    root: __dirname,
+    alias: config.aliases || {},
+    modules: config.moduleDirectories || [] // eg: ["node_modules", "node_modules_custom", "src"]
+  }
+}
+```
+
+### With package.json
+
+```js
+// webpack.config.js
+const config = require('./package.json')
+
+module.exports = {
+  entry: { ... },
+  resolve: {
+    root: __dirname,
+    alias: config.module-alias.aliases || {},
+    modules: config.module-alias.moduleDirectories || [] // eg: ["node_modules", "node_modules_custom", "src"]
+  }
+}
+```
+
+## 2.X Usage
```

Comment:
I think we need to rework the 2.X usage to remove everything that's duplicated from usage in 3.X .
Basically it needs to say 
- "you can't use a JS file, only a package.json, and this is what it should look like"
- "you can still configure the base path, for example..."
- "when using webpack, you need to do XYZ"

Remove everything that's related to advanced usage, because that's 100% identical to version 3.X

Reply:
Yes, this is a great idea.